File: 20241022/jueun/붕대감기.java
class Solution {
    public int solution(int[] bandage, int health, int[][] attacks) {        
        int hp = health;
        int streak = 1;
        
        int t = 1, T = attacks[attacks.length-1][0];
        int atk = 0;
        for(; t<=T; t++) {
            if(t == attacks[atk][0]) {  
                hp -= attacks[atk][1];
                
                if(hp <= 0)
                    return -1;
                
                atk++;
                streak = 1;
                
            } else if(streak == bandage[0]) {
                hp += bandage[2] + bandage[1];
                streak = 1;
            } else {
                hp += bandage[1];
                streak ++;
            }
            
            if(hp > health)
                hp = health;
            
            //System.out.printf("%d: %d %d\n", t, hp, streak-1);
        }

        return hp;
    }
}

File: 20241022/donggyu/붕대_감기.py
def recover(now_hp, bandage, health, continual):
    if continual == bandage[0]:
        recover = bandage[1]+bandage[2]
    else:
        recover = bandage[1]
        
    if now_hp + recover >= health:
        now_hp = health
    else:
        now_hp += recover

    return now_hp

def solution(bandage, health, attacks):
    answer = 0
    attack_t = attacks[-1][0]
    now_hp = health
    continual = 0
    attacks.reverse()
    
    for t in range(1, attack_t+1):
       #공격 매커니즘
        if t == attacks[-1][0]:
            now_hp -= attacks[-1][1]
            attacks.pop()
            continual = 1

            if now_hp <= 0:
                return -1            
            continue
            
        #회복 매커니즘
        now_hp = recover(now_hp, bandage, health, continual)    
        continual += 1
        if continual > bandage[0]:
            continual = 1

    return now_hp